cfme/tests/test_rest.py

```diff
@@ -374,13 +374,373 @@ def test_vm_add_lifecycle_event(request, setup_a_provider, rest_api, vm, from_de
+    policy_profile["policies"] = [{"href": _} for _ in added_policies]
+    rest_api.policy_profiles.action.add(policy_profile)
+    wait_for(
+        lambda: rest_api.policy_profiles.find_by(name=policy_profile.get("description")),
+        num_sec=180,
+        delay=10,
+    )
+
+    delete_policy_profile = rest_api.policy_profiles.find_by(policy_profile.get("description"))
+    delete_policy = delete_policy_profile.policies[0]
+    delete_policy.action.delete()
+    wait_for(
+        lambda: not delete_policy_profile.find_by(name=policy_profile.get("description")),
+        num_sec=180,
+        delay=10,
+    )
+
+    policies = [policy.id for policy in delete_policy_profile.policies]
+    rest_api.policy_profiles.delete(policies)
+    with error.expected("ActiveRecord::RecordNotFound"):
+        rest_api.policy_profiles.action.delete(policies)
+
+    delete_policy_profile.action.delete()
+    wait_for(
+        lambda: not rest_api.policy_profiles.find_by(name=policy_profile.get('description')),
+        num_sec=180,
+        delay=10,
+    )
+
+
+def test_add_delete_policy(added_policies, rest_api):
+    """Tests creating and deleting policy
+
+    Prerequisities:
+        * An appliance with ``/api`` available.
+
+    Steps:
+        * POST /api/policies (method ``add``)
+        * DELETE /api/polices/<id> -> delete only one policy
+        * Repeat the DELETE query -> now it should return an ``ActiveRecord::RecordNotFound``.
+
+        * DELETE /api/policies <- Insert a JSON with list of dicts containing ``href``s to
+            the policy
+        * Repeat the DELETE query -> now it should return an ``ActiveRecord::RecordNotFound``.
+
+    Metadata:
+        test_flag: rest
+    """
+    assert "add" in rest_api.policies.action and "delete" in rest_api.policies.action
+
+    delete_policy = rest_api.policies.find_by(id=added_policies[0])
+    delete_policy.action.delete()
+    wait_for(
+        lambda: not rest_api.policies.find_by(id=added_policies[0]),
+        num_sec=180,
+        delay=10,
+    )
+
+    policies = [policy for policy in rest_api.policies]
+    rest_api.policies.delete(policies)
+    with error.expected("ActiveRecord::RecordNotFound"):
+        rest_api.policies.action.delete(policies)
+
+
+def test_edit_policy(added_policies, rest_api):
+    """Tests editing a policy
+
+    Prerequisities:
+        * An appliance with ``/api`` available.
+
+    Steps:
+        * Retrieve list of entities using GET /api/policies , pick the first one
+        * POST /api/policies/1 (method ``edit``) with the ``description``
+
+    Metadata:
+        test_flag: rest
+    """
+    assert "edit" in rest_api.policies.action
+
+    policy = rest_api.policies[0]
+
+    new_description = "description_{}".format(fauxfactory.gen_alphanumeric())
+
+    policy.action.edit(description=new_description)
+    wait_for(
+        lambda: rest_api.policies.find_by(description=new_description),
+        num_sec=180,
+        delay=10,
+    )
```

Comment:
You could also use `fail_func=policy.reload` and then in the lambda  have just `lambda: policy.description == new_description` so it would utilize the original object, just reloading it by its id/href. But this approach is also good.